ext/web/lib.rs

```diff
@@ -149,8 +149,25 @@ fn op_base64_decode(
   _state: &mut OpState,
   input: String,
   _: (),
-) -> Result<ZeroCopyBuf, AnyError> {
-  let mut input: &str = &input.replace(|c| char::is_ascii_whitespace(&c), "");
+) -> Result<String, AnyError> {
+  let mut ws = false;
+  let len = input.len();
+  let invalid = input.char_indices().any(|(i, c)| {
+    if char::is_ascii_whitespace(&c) {
+      ws = true;
+    } else if c == '=' {
+      if i != len - 1 && i != len - 2 {
+        return true;
+      }
+    } else if c != '+' && c != '/' && !c.is_alphanumeric() {
+      return true;
+    }
+    false
+  });
+
+  // "Remove all ASCII whitespace from data"
+  let input = if ws { input.replace(|c| char::is_ascii_whitespace(&c), "") } else { input };
```

Comment:
It would be preferable if the replacement could be done on the same iter as the invalid characters but that's probably not possible.